hclfmt

yorinasub17 opened the pull request:
According to https://github.com/hashicorp/terraform/pull/22092, it seems unlikely that `terraform fmt` would support formatting `terragrunt.hcl` directly. In that PR, it was suggested that [we should standardize to using `hclfmt`](https://github.com/hashicorp/terraform/pull/22092#issuecomment-512028847). However, this is an additional utility that one needs to install, not to mention that it is not officially released yet.

As such, this PR bakes the functionality of `hclfmt` as a sub command of `terragrunt`, so that you can run `terragrunt hclfmt` to format your `terragrunt.hcl` files in the directory tree.

Reference: https://github.com/gruntwork-io/terragrunt/issues/768

brikis98 commented on cli/cli_app.go:
```diff
@@ -308,6 +320,13 @@ func shouldPrintTerragruntInfo(terragruntOptions *options.TerragruntOptions) boo
+	if util.ListContainsElement(terragruntOptions.TerraformCliArgs, CMD_HCLFMT) {
+		return true
+	}
+	return false
```
Simplify to `return util.ListContainsElement(...)`

brikis98 commented on cli/hclfmt.go:
```diff
@@ -0,0 +1,115 @@
+}
+
+// findTerragruntHclFiles will recursively look for terragrunt.hcl files in the directory tree starting at workingDir.
+func findTerragruntHclFiles(terragruntOptions *options.TerragruntOptions, workingDir string) ([]string, error) {
```
Alternative option is to use the [zglob library](https://github.com/mattn/go-zglob) to give back a list in a one-liner.

brikis98 commented on cli/hclfmt.go:
```diff
@@ -0,0 +1,115 @@
+	if err != nil {
+		return nil, err
+	}
+	return ioutil.ReadAll(f)
```
https://github.com/gruntwork-io/terragrunt/blob/master/util/file.go#L131

yorinasub17 commented:
UPDATE:

- Simplified `shouldRunHCLFmt` and `shouldPrintTerragruntInfo`
- Replaced `findTerragruntHclFiles` with `zglob.Glob`
- Replaced `readAllFromFile` with `util.ReadFileAsString`
- Updated working directory to pull from `options.WorkingDir`

yorinasub17 commented:
Ok going to merge + release this now. Thanks for the review!
